accounting/accounting.go

```diff
@@ -0,0 +1,188 @@
+package accounting
+
+import (
+	"math"
+	"math/big"
+	"strconv"
+	"strings"
+
+	"github.com/LUSHDigital/core-lush/currency"
+)
+
+const (
+	// QuadruplePrecision describes 128 bits of precision for IEEE 754 decimals
+	// see: https://en.wikipedia.org/wiki/Quadruple-precision_floating-point_format
+	QuadruplePrecision uint = 128
+
+	// OctuplePrecision describes 256 bits of precision for for IEEE 754 decimals
+	// see: https://en.wikipedia.org/wiki/Octuple-precision_floating-point_format
+	OctuplePrecision uint = 256
+)
+
+var (
+	// Minimum rate of tax must be 0.
+	// As far as we know, there are as of writing, no markets with a negative sales tax.
+	min = itof(0)
+
+	// Baseline for creating a rate divisor.
+	base = itof(1)
+)
+
+// ValidateManyFloatsArePrecise tests that the given float64 arguments
+// have the desired precision, this is a convenience wrapper
+// around ValidateFloatIsPrecise.
+// NOTE: 2 digits past the dot is a business rule.
+func ValidateManyFloatsArePrecise(args ...float64) error {
+	for _, f := range args {
+		if err := ValidateFloatIsPrecise(f); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+// ValidateFloatIsPrecise ensures that a float64 value does not exceed
+// a precision of 2 digits past the dot. This ensures we do not
+// store incorrect currency data.
+// NOTE: 2 digits past the dot is a business rule.
```

Comment:
For clarity should this be `decimal separator` rather than `dot`? see https://en.wikipedia.org/wiki/Decimal_separator

Reply:
more specifically we only use the decimal period.
so I will update to reflect that!